nautobot/dcim/tests/test_filters.py

```diff
@@ -2924,3 +2974,236 @@ def test_failover_strategy(self):
 
 
 # TODO: Connection filters
+
+
+class InterfaceRedundancyGroupTestCase(FilterTestCases.FilterTestCase):
+    queryset = InterfaceRedundancyGroup.objects.all()
+    filterset = InterfaceRedundancyGroupFilterSet
+
+    @classmethod
+    def setUpTestData(cls):
+        common_test_data(cls)
+
+        statuses = Status.objects.get_for_model(InterfaceRedundancyGroup)
+        cls.ips = IPAddress.objects.all()
+
+        interface_redundancy_groups = (
+            InterfaceRedundancyGroup(
+                name="Interface Redundancy Group 1",
+                protocol="hsrp",
+                protocol_group_id="1",
+                status=statuses[0],
+                virtual_ip=cls.ips[0],
+            ),
+            InterfaceRedundancyGroup(
+                name="Interface Redundancy Group 2",
+                protocol="carp",
+                protocol_group_id="2",
+                status=statuses[1],
+                virtual_ip=cls.ips[1],
+            ),
+            InterfaceRedundancyGroup(
+                name="Interface Redundancy Group 3",
+                protocol="vrrp",
+                protocol_group_id="3",
+                status=statuses[2],
+                virtual_ip=cls.ips[2],
+            ),
+            InterfaceRedundancyGroup(
+                name="Interface Redundancy Group 4",
+                protocol="glbp",
+                protocol_group_id="4",
+                status=statuses[3],
+                virtual_ip=cls.ips[3],
+            ),
+        )
+        tags = Tag.objects.get_for_model(InterfaceRedundancyGroup)
+        for group in interface_redundancy_groups:
+            group.tags.set(tags)
+            group.validated_save()
+
+        secrets_groups = list(SecretsGroup.objects.all()[:2])
+
+        interface_redundancy_groups[0].secrets_group = secrets_groups[0]
+        interface_redundancy_groups[0].validated_save()
+
+        interface_redundancy_groups[1].secrets_group = secrets_groups[1]
+        interface_redundancy_groups[1].validated_save()
+
+    def test_name(self):
```

Comment:
Would be good to rework all of these filter tests into the 2.0 filter testing paradigm (`generic_filter_tests`, `assertQuerySetEqualAndNotEmpty`, etc.). Not blocking.